logger/analytics/analyticslogger.go

```diff
@@ -172,25 +206,14 @@ func (al *Logger) Write(bs []byte) (int, error) {
 			}
 		}()
 	}
-	al.mu.Unlock()
-	return len(bs), nil
+	return nil
 }
 
 // Close flushes all logs to Firehose.
 func (al *Logger) Close() error {
-	al.mu.Lock()
-	if len(al.batch) > 0 {
-		batch := al.batch
-		// be careful not to send al.batch, since we will unlock before we finish sending the batch
-		err := sendBatch(batch, al.fhAPI, al.fhStream, time.Now().Add(timeoutForSendingBatches))
-		if err != nil {
-			al.errLogger.ErrorD("send-batch-error", logger.M{
-				"stream": al.fhStream,
-				"error":  err.Error(),
-			})
-		}
-	}
-	al.mu.Unlock()
+	al.sendingTicker.Stop()
```

Comment:
Just thinking out loud about what happens if `Close()` gets called multiple times. It seems safe to me.
* `done` is an unbuffered/no-max-capacity channel that never gets closed. At worst, some `true` values go into the void because the background routine already stopped.
* It seems safe to call `Stop()` on an already stopped `Ticker`, and similarly `Wait()` multiple times on a `WaitGroup`

Reply:
It seems like there's some text about reusing wait groups at https://golang.org/pkg/sync/#WaitGroup.Add

Reply:
Good callout, but agreed it seems safe. I'm a little confused about the note there, but I _think_ the point is just that if you call `Wait()` and then `Add()` then you'll have to call `Wait()` again if you want it to wait on the newly added thing (which in this case we are doing since `Wait()` is at the end).